mongo/src/main/java/com/redhat/lightblue/mongo/crud/js/JSQueryTranslator.java

```diff
@@ -219,20 +229,25 @@ private Block translateFieldComparison(Context ctx,FieldComparisonExpression que
             Name arrayFieldName;
             Name simpleFieldName;
             BinaryComparisonOperator op;
+            boolean isDate;
             if(rFieldMd instanceof ArrayField) {
                 arrayFieldName=rfieldName;
                 simpleFieldName=lfieldName;
                 op=query.getOp().invert();
+                isDate=((ArrayField)rFieldMd).getElement().getType() instanceof DateType && lFieldMd.getType() instanceof DateType;
             } else {
                 arrayFieldName=lfieldName;
                 simpleFieldName=rfieldName;
                 op=query.getOp();
+                isDate=((ArrayField)lFieldMd).getElement().getType() instanceof DateType && rFieldMd.getType() instanceof DateType;
```

Comment:
At this point, only one of lField or rField is an ArrayField. The if statement is for rField being an array, and this is in the else, so at this point, lField is an array.